bgflow/distribution/sampling/base.py

```diff
@@ -1,13 +1,25 @@
 import torch
-
+from ...utils.types import unpack_tensor_tuple, pack_tensor_in_list
 
 __all__ = ["Sampler"]
 
 
 class Sampler(torch.nn.Module):
```

Comment:
This now has a `return_hook` to do something with the samples (e.g. reshaping or selecting elements) before they are returned.